frontend/src/components/MailItem.tsx

```diff
@@ -50,6 +59,45 @@ export default function MailItem({
   const [visible, setVisible] = React.useState<boolean>(false)
   const [showConfirmRemove, setConfirmRemove] = React.useState<boolean>(false)
   const video_date = new Date(created_at)
+  const [
+    profileContent,
+    setProfileContent,
+  ] = React.useState<ProfileContentType>({
+    first_name: "",
+    last_name: "",
+    birth_date: null,
+    gender: null,
+    interests: [],
+    image: null,
+    video: null,
+  })
+  const [showProfile, setShowProfile] = React.useState<boolean>(false)
+
+  React.useEffect(() => {
+    request<ProfileContentType>({ path: `/api/profiles/${id}/`, method: "get" })
+      .then((res) => {
+        const profile_content = res.parsedBody
+
+        // Fields that are restricted on a user's profile (i.e. Private or Hidden)
+        // are not returned for privacy purposes. Test if certain field were not
+        // returned to avoid frontend errors related to undefined values.
+        if (typeof profile_content.birth_date === "undefined") {
```

Comment:
One thing I found was the birthday of new users was set automatically which is weird. Also, the bug from my last comment is still there. I set the first and last name of a match to hidden and it didn't hide those demographics which were right next to the "view profile" button.

Reply:
that would hide the last name of a penpal from the mail page if set to hidden